src/store/profile.store.ts

```diff
@@ -0,0 +1,72 @@
+import { createSlice, Selector, PayloadAction } from '@reduxjs/toolkit';
+import { RootState } from './app.store';
+
+export interface ProfileState {
+  teachingPercent: number;
+  researchPercent: number;
+  servicePercent: number;
+}
+
+const initialState: ProfileState = {
+  teachingPercent: 0,
+  researchPercent: 0,
+  servicePercent: 0,
+};
+
+export const profileSlice = createSlice({
+  name: 'Profile',
+  initialState,
+  reducers: {
+    setTeachingPercent: (state, action: PayloadAction<number>) => {
+      state.teachingPercent = action.payload;
+    },
+    setResearchPercent: (state, action: PayloadAction<number>) => {
+      state.researchPercent = action.payload;
+    },
+    setServicePercent: (state, action: PayloadAction<number>) => {
+      state.servicePercent = action.payload;
+    },
+    setPercent: (
+      state,
+      action: PayloadAction<{ type: string; percent: number }>,
+    ) => {
+      switch (action.payload.type) {
+        case 'teaching':
+          state.teachingPercent = action.payload.percent;
+          break;
+        case 'creative':
```

Comment:
not needed?

Reply:
It's a weird reason why, but it is needed; `setPercent` is called in PercentageInfo line 51, and it uses the category label (one of "Teaching", "Creative", or "Service") as the field. so we need to include both "creative" and "research" as we use the names interchangeably (need to handle both). Different solution would be to change the label from "Creative" to "Research", or to modify the `labels` object to have both a label ("Creative") and a field ("research").

Reply:
I see so creative and research will set the reserachPercent